examples/with-wallet-stamper/src/components/turnkey-provider.tsx

```diff
@@ -131,37 +212,128 @@ export const TurnkeyProvider: React.FC<TurnkeyProviderProps> = ({
-    const { organizationIds } = await getSubOrgByPublicKey(publicKey);
-    const organizationId = organizationIds[0];
-
+  // Sign in with wallet and create an active session
+  async function signInWithWallet(email?: Email): Promise<User | null> {
     let whoami: User | null = null;
-    if (walletClient) {
-      try {
-        whoami = await walletClient.getWhoami({
-          organizationId,
-        });
-        setUser(whoami);
-        router.push("/dashboard");
-      } catch (e) {
-        console.error(e);
+
+    if (!walletClient) {
+      return null;
+    }
+
+    // If no email is provided, log in with wallet. Will require multiple approvals.
+    if (!email) {
+      const publicKey = (wallet as SolanaWalletInterface)?.recoverPublicKey();
+      const { organizationIds } = await getSubOrgByPublicKey(publicKey); // One wallet request
+      const organizationId = organizationIds[0];
+
+      if (walletClient) {
+        try {
+          // Another wallet request
+          whoami = await walletClient.getWhoami({
+            organizationId,
+          });
+          setUser(whoami);
+          router.push("/dashboard");
+        } catch (e) {
+          console.error(e);
+        }
       }
+
+      return whoami;
     }
+
+    // Create an iframe session. Email is required for the next bit.
+    try {
+      const activityPoller = createActivityPoller({
+        client: walletClient,
+        requestFn: walletClient.createReadWriteSession,
+      });
+
+      const completedActivity = await activityPoller({
+        type: "ACTIVITY_TYPE_CREATE_READ_WRITE_SESSION",
+        timestampMs: new Date().getTime().toString(),
+        organizationId: NEXT_PUBLIC_ORGANIZATION_ID, // This can accept the parent org if we don't know the suborg in advance
+        parameters: {
+          email: email!,
+          targetPublicKey: iframeClient?.iframePublicKey!,
+        },
+      });
+
+      const result = completedActivity.result.createReadWriteSessionResult;
+
+      // Need to save resulting bundle and information.
+      // This is required by the authIframeClient abstraction.
+      localStorage.setItem("CREDENTIAL_BUNDLE", result?.credentialBundle!);
+      localStorage.setItem(
+        "@turnkey/current_user",
+        JSON.stringify({
+          userId: result?.userId,
+          username: result?.username,
+          organization: {
+            organizationId: result?.organizationId,
+            organizationName: result?.organizationName,
+          },
+        })
+      );
+
+      setUser({
+        userId: result?.userId!,
+        username: result?.username!,
+        organizationId: result?.organizationId!,
+        organizationName: result?.organizationName!,
+      });
+      router.push("/dashboard");
+    } catch (e) {
+      console.error(e);
+    }
+
     return whoami;
   }
 
-  async function createWallet(walletName: string) {
+  async function createWallet(walletName: string, callback: any) {
```

Comment:
Note: bit of an antipattern. See caller in `examples/with-wallet-stamper/src/components/user.tsx`